packages/components/icon/src/Icon.tsx

```diff
@@ -0,0 +1,142 @@
+import { css, cx } from 'emotion';
+import React, { forwardRef } from 'react';
+import tokens from '@contentful/f36-tokens';
+import { Primitive } from '@contentful/f36-core';
+import type {
+  CommonProps,
+  ComponentVariant,
+  Simplify,
+} from '@contentful/f36-core';
+import type {
+  ComponentType,
+  ExoticComponent,
+  ReactElement,
+  SVGAttributes,
+} from 'react';
+
+export type IconComponent = ExoticComponent<any> | ComponentType<any>;
+
+type IconVariant = Simplify<ComponentVariant | 'muted' | 'white'>;
+
+type AsOrChildren =
+  | {
+      as: ExoticComponent<any> | ComponentType<any>;
+      children?: never;
+    }
+  | {
+      as?: never;
+      children: ReactElement | ReactElement[];
+    };
+
+type IconInternalProps = {
+  /**
+   * The name prop is used to check if "trimmed" styling should be applied,
+   * which sets the `width` to `auto`.
+   */
+  name?: string;
+  /**
+   * Determines the size of the icon
+   */
+  size?: 'large' | 'medium' | 'small' | 'tiny';
+  /**
+   * Determines the fill color used
+   */
+  variant?: IconVariant;
+  /**
+   * SVG viewBox attribute
+   */
+  viewBox?: `${number} ${number} ${number} ${number}`;
```

Comment:
this type won't work for Typescript versions before 4.2, in this particular case I'd simply fallback to `viewBox?: string`

Reply:
As far as I know we're requiring TS v4+ for our own v4. Also, there should already be a string fallback added by the compiler itself, at least that how the type works for me.

Reply:
I think we should aim for Typescript@3.8+ for our users. And this Template syntax won't work with 4.0, 4.1, so I would rather be safe here

Reply:
It's only the 4.0 upgrade that has breaking changes. In any case, if we target 3.8 I think we should use that ourselves.